### PR TITLE
Documentation fixes.

### PR DESCRIPTION
**Pull request by awulkiew**
1. Wrong arguments' names of `discrete_hausdorff_distance()`
2. Wrong description of `distance()`
https://github.com/boostorg/geometry/issues/521
3. Missing documentation of `dsv()`
https://github.com/boostorg/geometry/issues/520

### CODE REVIEW

File: include/boost/geometry/algorithms/detail/distance/interface.hpp [hunk 1]
```diff
@@ -320,10 +321,10 @@ struct distance
 
 
 /*!
-\brief \brief_calc2{distance} \brief_strategy
+\brief Calculate the distance between two geometries \brief_strategy
 \ingroup distance
 \details
-\details \details_calc{area}. \brief_strategy. \details_strategy_reasons
+\details The free function distance calculates the distance between two geometries \brief_strategy. \details_strategy_reasons
```
Comment (mloskot) [inline include/boost/geometry/algorithms/detail/distance/interface.hpp]: Shouldn't it read

> The free `distance` function...?

File: include/boost/geometry/algorithms/detail/distance/interface.hpp [hunk 2]
```diff
@@ -376,9 +377,10 @@ distance(Geometry1 const& geometry1,
 
 
 /*!
-\brief \brief_calc2{distance}
+\brief Calculate the distance between two geometries.
 \ingroup distance
-\details The default strategy is used, corresponding to the coordinate system of the geometries
+\details The free function distance calculates the distance between two geometries. \details_default_strategy
```
Comment (mloskot) [inline include/boost/geometry/algorithms/detail/distance/interface.hpp]: Shouldn't it read

> The free `distance` function calculates...?

File: include/boost/geometry/algorithms/discrete_hausdorff_distance.hpp [hunk 3]
```diff
@@ -264,7 +264,9 @@ struct discrete_hausdorff_distance<multi_linestring1, multi_linestring2, multi_l
 // Algorithm overload using explicitly passed Pt-Pt distance strategy
 
 /*!
-\brief calculate discrete hausdorff distance between two geometries ( currently works for LineString-LineString, MultiPoint-MultiPoint, Point-MultiPoint, MultiLineString-MultiLineString ) using specified strategy
+\brief Calculate discrete hausdorff distance between two geometries (currently
```
Comment (mloskot) [inline include/boost/geometry/algorithms/discrete_hausdorff_distance.hpp]: I think it should be capitalised  `Hausdorff`

File: include/boost/geometry/algorithms/discrete_hausdorff_distance.hpp [hunk 4]
```diff
@@ -295,13 +297,20 @@ typename distance_result
...
-    return dispatch::discrete_hausdorff_distance<Geometry1, Geometry2>::apply(g1, g2, strategy);
+    return dispatch::discrete_hausdorff_distance
+        <
+            Geometry1, Geometry2
+        >::apply(geometry1, geometry2, strategy);
 }
 
 /*!
-\brief calculate discrete hausdorff distance between two geometries ( currently works for LineString-LineString, MultiPoint-MultiPoint, Point-MultiPoint, MultiLineString-MultiLineString )
+\brief Calculate discrete hausdorff distance between two geometries (currently
```
Comment (mloskot) [inline include/boost/geometry/algorithms/discrete_hausdorff_distance.hpp]: I think it should be capitalised  `Hausdorff`

**Comment by mloskot** [comment]
LGTM, just posted a few minor comments. Thanks!

File: include/boost/geometry/algorithms/detail/distance/interface.hpp (hunk 1 above)
Comment (awulkiew) [reply include/boost/geometry/algorithms/detail/distance/interface.hpp]: It's from a template we use:
https://github.com/boostorg/geometry/blob/develop/doc/doxy/Doxyfile#L98

File: include/boost/geometry/algorithms/discrete_hausdorff_distance.hpp (hunk 3 above)
Comment (awulkiew) [reply include/boost/geometry/algorithms/discrete_hausdorff_distance.hpp]: Right, thanks.

Comment (awulkiew) [reply include/boost/geometry/algorithms/discrete_hausdorff_distance.hpp]: I also fixed Frechet distance the same way.

File: include/boost/geometry/algorithms/detail/distance/interface.hpp (hunk 1 above)
Comment (awulkiew) [reply include/boost/geometry/algorithms/detail/distance/interface.hpp]: Ah, indeed there is repetition. How about something like `The free function distance calculates the length of the shortest path between two geometries`?

Comment (mloskot) [reply include/boost/geometry/algorithms/detail/distance/interface.hpp]: I think it's good, for hand-written sentences. But, for the template, I think the current template is acceptable.

**Pull request merged**